go/connector/run.go

```diff
@@ -293,7 +293,7 @@ func runCommand(
 		var cidFileReadDeadline = time.Now().Add(time.Second * 5)
 		for {
 			idBytes, err := os.ReadFile(cidFilePath)
-			if err == nil {
+			if err == nil && len(idBytes) > 0 {
```

Comment:
Is there a way to pick a random port on the host and then explicitly map that above on line 160 when the container is started? That seems like it would simplify needing to query/retry the container to see which port has been randomly chosen.

Reply:
@williamhbaker I've been thinking about this too... I just found this: https://github.com/phayes/freeport

This could be a great simplification to be honest, I don't like what is happening here with all of these retries, etc... 🤔 
I'll try that